superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.jsx

```diff
@@ -51,98 +51,49 @@ const StyledScrollbarContent = styled.div`
   height: ${props => props.contentHeight}px;
 `;
 
-export default class SqlEditorLeftBar extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.resetState = this.resetState.bind(this);
-    this.onSchemaChange = this.onSchemaChange.bind(this);
-    this.onSchemasLoad = this.onSchemasLoad.bind(this);
-    this.onTablesLoad = this.onTablesLoad.bind(this);
-    this.onDbChange = this.onDbChange.bind(this);
-    this.getDbList = this.getDbList.bind(this);
-    this.onTableChange = this.onTableChange.bind(this);
-    this.onToggleTable = this.onToggleTable.bind(this);
-  }
-
-  onSchemaChange(schema) {
-    this.props.actions.queryEditorSetSchema(this.props.queryEditor, schema);
-  }
-
-  onSchemasLoad(schemas) {
-    this.props.actions.queryEditorSetSchemaOptions(
-      this.props.queryEditor,
-      schemas,
-    );
-  }
-
-  onTablesLoad(tables) {
-    this.props.actions.queryEditorSetTableOptions(
-      this.props.queryEditor,
-      tables,
-    );
-  }
-
-  onDbChange(db) {
-    this.props.actions.queryEditorSetDb(this.props.queryEditor, db.id);
-    this.props.actions.queryEditorSetFunctionNames(
-      this.props.queryEditor,
-      db.id,
-    );
-  }
-
-  onTableChange(tableName, schemaName) {
+export default function SqlEditorLeftBar(props) {
```

Comment:
I would suggest as an improvement here, to destructure the props into each individual property. Just one level is probably fine, like `actions`, `tables`, `queryEditor`, etc. That will clean up a lot of the `props.` code in this file.